qodana.yaml

```diff
@@ -0,0 +1,31 @@
+#-------------------------------------------------------------------------------#
```

Comment:
What would proper integration entail? I removed qodana.yaml for now and added to our .gitignore. See below commits.

Reply:
I'd say something that can be run with maven easily like already happening with Checkstyle so it's the same reproducible workflow for every developer
Otherwise having people just commit their favorite tool configs doesn't seem very useful

Reply:
Apparently, I can use IDEA but not Eclipse (see ask-devs discord channel). IDEA has "Server-Side Analysis". Setting this up created qodana.yaml. It seems useful to enable all IDEA capabilities.

How do I use Checkstyle?

I have removed qodana.yaml from our .gitignore.

Are we good to go?

Reply:
Checkstyle offers plugins for different IDE
but like I said it's already assigned a maven phase too
I don't use IntelliJ + looks like we already have a Findbugs config too (I know it's outdated)
But we can certainly consider upgrading to a better linter, though preferably something more open source...